File: Difficulty/Easy/[38] Count and Say/solution.py
#
# @lc app=leetcode id=38 lang=python3
#
# [38] Count and Say
#
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return '1'
        if n == 2:
            return '11'
        s = '11'
        for _ in range(n - 2):
            temp, num, c = '', 0, ''
            for elem in s:
                if c and elem == c:
                    num += 1
                elif c and elem != c:
                    temp += str(num) + c
                    num = 1
                    c = elem
                else:
                    num = 1
                    c = elem
            temp += str(num) + c
            s = temp
        return s

File: Difficulty/Easy/[38] Count and Say/solution.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n == 1)
            return "1";
        if (n == 2)
            return "11";
        string s = "11";
        for (int i {}; i < n - 2; i++){
            string temp = "";
            char c = 0;
            int num = 0;
            for (auto it = s.begin(); it != s.end(); it++){
                if (c && *it == c)
                    num += 1;
                else if (c && *it != c){
                    temp += to_string(num) + c;
                    num = 1;
                    c = *it;
                }
                else{
                    num = 1;
                    c = *it;
                }
            }
            temp += to_string(num) + c;
            s = temp;
        }
        return s;
    }
};